src/data/finsupp/basic.lean

```diff
@@ -356,6 +356,44 @@ by { ext, simp [finsupp.single_eq_pi_single, finsupp.equiv_fun_on_fintype], }
 
 end single
 
+/-! ### Declarations about `update` -/
+
+section update
+
+variables [decidable_eq α] [has_zero M] (f : α →₀ M) (a : α) (b : M) [decidable (b = 0)] (i : α)
```

Comment:
I understand the motivation here, but I think finsupp has committed to not having this sort of decidability argument and going non-computable instead. `dfinsupp` is the version that does take the approach you use here

Reply:
What do you suggest is done here? That this def + lemmas are added to `dfinsupp` instead? Since I use this def in #9020, what's the right way to phrase it?

Reply:
Ideally we'd have both finsupp.update and dfinsupp.update. My hunch is that in this definition you should be using a classical decidable instance for at least `b=0`. Perhaps @digama0 has an opinion.

Reply:
Yes, just drop the `decidable` hypotheses and go classical/noncomputable as necessary.